components/dashboard/src/components/IdeOptions.tsx

```diff
@@ -0,0 +1,283 @@
+    const ideOptionsArr = useMemo(() => (ideOptions ? sortedIdeOptions(ideOptions) : undefined), [ideOptions]);
+    const pinnableIdes = useMemo(() => ideOptionsArr?.filter((i) => !!i.pinnable), [ideOptionsArr]);
+
+    const [restrictedEditors, setEditors] = useState(props.restrictedEditors);
+    const [pinnedEditorVersions, setPinnedEditorVersions] = useState(props.pinnedEditorVersions);
+    const [editorVersions, setEditorVersions] = useState(new Map<string, string[]>());
+
+    useEffect(() => {
+        async function fetchData() {
+            if (!pinnableIdes) {
+                return;
+            }
+            const ideVersionsResult = await Promise.all(
+                pinnableIdes.map((ide) => getGitpodService().server.getIDEVersions(ide.id)),
+            );
+            const updatedVal = new Map<string, string[]>();
+            for (let i = 0; i < pinnableIdes.length; i++) {
+                const versions = ideVersionsResult[i]!;
+                updatedVal.set(pinnableIdes[i].id, versions);
+            }
+            setEditorVersions(updatedVal);
+        }
+        fetchData();
+    }, [pinnableIdes]);
+
+    const { toast } = useToast();
+
+    const handleUpdate = async () => {
+        if (computedError) {
+            return;
+        }
+        updateMutation.mutate(
+            { restrictedEditors, pinnedEditorVersions },
+            {
+                onSuccess: () => {
+                    toast({ message: "Editor options updated" });
+                    onClose();
+                },
+            },
+        );
+    };
+
+    const computedError = useMemo(() => {
+        if (!ideOptionsArr) {
+            return;
+        }
+        const leftOptions = ideOptionsArr.filter((i) => !restrictedEditors.has(i.id));
+        if (leftOptions.length === 0) {
+            return "At least one Editor has to be selected.";
+        }
+    }, [restrictedEditors, ideOptionsArr]);
+
+    const isLoading = props.isLoading || !pinnableIdes || pinnableIdes.length !== editorVersions.size;
+
+    return (
+        <Modal visible onClose={onClose} onSubmit={handleUpdate}>
+            <ModalHeader>Available Editors</ModalHeader>
+            <ModalBody>
+                {isLoading ? (
+                    <LoadingState />
+                ) : (
+                    ideOptionsArr!.map((ide) => (
+                        <IdeOptionSwitch
+                            key={ide.id}
+                            ideOption={ide}
+                            ideVersions={editorVersions.get(ide.id)}
+                            pinnedIdeVersion={pinnedEditorVersions.get(ide.id)}
+                            checked={!restrictedEditors.has(ide.id)}
+                            onPinnedIdeVersionChange={(version) => {
+                                const updatedVal = new Map(pinnedEditorVersions);
+                                !!version ? updatedVal.set(ide.id, version) : updatedVal.delete(ide.id);
```

Comment:
Implicitly casted as a bool here
```suggestion
                                version ? updatedVal.set(ide.id, version) : updatedVal.delete(ide.id);
```